💡 TTS 小样本 finetune / 声音克隆问题汇总

**如果 12 句 finetune 效果不佳，一般是因为数据集太小了，建议增加数据集，一般是 300 ~ 600 条，数据量和质量越好，合成的效果越好**
数据的质量要求没有混响，没有杂音，离麦克风距离适中，具体可以参考标贝的数据质量。
finetune 出来的音色与 目标说话人和原始说话人的相似度有关，即目标说话人和原始说话人相似度越高，finetune 出来的音色更接近目标说话人。
finetune 出来的音频质量与原始说话人的音频质量有关，原始说话人的音频质量不好，finetune 出来的效果也可能不好。
综上，finetune 方案在数据采集，选择原始说话人上需要好好选择。

小样本 finetune 原理参考 [关于训练一个自己的TTS模型](https://github.com/PaddlePaddle/PaddleSpeech/discussions/1842)

1. https://github.com/PaddlePaddle/PaddleSpeech/issues/2437
2. https://github.com/PaddlePaddle/PaddleSpeech/issues/2454
3. https://github.com/PaddlePaddle/PaddleSpeech/issues/2383
4. 预处理都没有问题，为什么不跑训练流程？-> epoch 的设置有问题，参考： https://github.com/PaddlePaddle/PaddleSpeech/issues/2319#issuecomment-1231618015
5. https://github.com/PaddlePaddle/PaddleSpeech/issues/2442
6. https://github.com/PaddlePaddle/PaddleSpeech/issues/2471 -> 安装 develop 版本的 paddlespeech
7. https://github.com/PaddlePaddle/PaddleSpeech/issues/2245
8. https://github.com/PaddlePaddle/PaddleSpeech/issues/2485 -> 安装 develop 版本的 paddlespeech
9. https://github.com/PaddlePaddle/PaddleSpeech/issues/2583 -> 推荐使用 finetune 方案
10. https://github.com/PaddlePaddle/PaddleSpeech/issues/2586
11. https://github.com/PaddlePaddle/PaddleSpeech/issues/2607
12. https://github.com/PaddlePaddle/PaddleSpeech/issues/2790
13. https://github.com/PaddlePaddle/PaddleSpeech/issues/2953

./run.sh --stage 0 --stop-stage 5
check oov
get mfa result
sh: 1: mfa_align: Exec format error
generate durations.txt
extract feature
[nltk_data] Error loading averaged_perceptron_tagger: <urlopen error
[nltk_data]     [Errno 111] Connection refused>
[nltk_data] Error loading cmudict: <urlopen error [Errno 111]
[nltk_data]     Connection refused>
196 1
100%|███████████████████████████████████████████████████████████████████████████████████| 196/196 [00:00<00:00, 5146.26it/s]
Done
Traceback (most recent call last):
  File "local/extract_feature.py", line 346, in <module>
    extract_feature(
  File "local/extract_feature.py", line 266, in extract_feature
    normalize(speech_scaler, pitch_scaler, energy_scaler, vocab_phones,
  File "local/extract_feature.py", line 155, in normalize
    dataset = DataTable(
  File "/home/nx/study/python/Paddle24/PaddleSpeech/paddlespeech/t2s/datasets/data_table.py", line 45, in __init__
    assert len(data) > 0, "This dataset has no examples"
AssertionError: This dataset has no examples

The code in File "/home/nx/study/python/Paddle24/PaddleSpeech/paddlespeech/t2s/datasets/data_table.py", line 45:
        _self.data = data
        assert len(data) > 0, "This dataset has no examples"_

--- a/docs/requirements.txt
+++ b/docs/requirements.txt
@@ -23,7 +23,7 @@ paddlespeech_feat
 pandas
 pattern_singleton
 ppdiffusers>=0.9.0
-praatio>=5.0.0
+praatio>=5.0.0, <=5.1.1
 prettytable
 pypinyin-dict
 pypinyin<=0.44.0

--- a/setup.py
+++ b/setup.py
@@ -32,10 +32,12 @@ VERSION = '0.0.0'
 COMMITID = 'none'
 
 base = [
+    "braceexpand",
     "editdistance",
     "g2p_en",
     "g2pM",
     "h5py",
+    "hyperpyyaml",
     "inflect",
     "jsonlines",
     "librosa==0.8.1",
@@ -46,14 +48,18 @@ base = [
     "opencc",
     "opencc-python-reimplemented",
     "pandas",
+    "paddleaudio>=1.1.0",
     "paddlenlp>=2.4.8",
+    "paddleslim>=2.3.4",
     "ppdiffusers>=0.9.0",
     "paddlespeech_feat",
     "praatio>=5.0.0, <=5.1.1",
+    "prettytable",
     "pypinyin<=0.44.0",
     "pypinyin-dict",
     "python-dateutil",
     "pyworld>=0.2.12",
+    "pyyaml",
     "resampy",
     "sacrebleu",
     "textgrid",
@@ -62,13 +68,7 @@ base = [
     "typeguard",
     "webrtcvad",
     "yacs~=0.1.8",
-    "prettytable",
     "zhon",
-    "braceexpand",
-    "pyyaml",
-    "paddleslim>=2.3.4",
-    "paddleaudio>=1.1.0",
-    "hyperpyyaml",
 ]
 
 server = ["pattern_singleton", "websockets"]

--- a/setup.py
+++ b/setup.py
@@ -49,7 +49,7 @@ base = [
     "paddlenlp>=2.4.8",
     "ppdiffusers>=0.9.0",
     "paddlespeech_feat",
-    "praatio>=5.0.0",
+    "praatio>=5.0.0, <=5.1.1",
     "pypinyin<=0.44.0",
     "pypinyin-dict",
     "python-dateutil",

